### ISSUE TITLE
[Spell] Hunter's Mark and Kill Shot

### ISSUE THREAD

**Issue by Dranah** [WoWManiaUK/Redemption#2836]
http://www.wow-mania.com/armory/?spell=53338

I found this while digging in hunter calculations for the other report.

**How it should work:** Hunter's mark adds 500 RAP against the marked target. This means the effect should always behave and grant damage as if the hunter received a +500 AP buff, towards that one target.

**How it works atm:** Hunter's mark does not add as much damage as 500 RAP would suggest for Kill shot. The spell somehow calculates wrongly with the extra RAP.
Other spells I tested (auto shot, aimed shot, arcane shot) looked alright, I suppose it's Kill Shot's formula that is using the mark's bonus wrong.

**TESTS**
Kill shot damage= 200% weapon damage + 650 (base) + 0,4 x RAP
I compared the damage of Kill Shot with 500 RAP difference from gear, or 500 RAP difference from hunter's mark. Result should be same but isn't. (for other spells , it's same as intended)
All tests were done with 100% arpen by gear = 9,2% mitigation left on target (Lord Marrowgar)

TEST 1 : hunter with 4340 RAP against unmarked target
Weapon damage (displayed) 1915-2204
Expected damage: 5644- 6168
Tested damage:5656-6150
Looks good.

TEST 2: hunter with 4840 RAP against unmarked target
Weapon damage: 2022- 2311
Expected damage: 6020- 6544
Tested damage: 6052- 6534
Looks ok.

TEST 3: hunter with 4340 RAP against marked target (so effective RAP 4840)
Expected damage: 6020-6544 (same as test2)
Tested damage: 5742- 6258
This is totally off, I could not find math that would explain these numbers. 
The damage should match with TEST 2, because the effective RAP should be same, but it's much closer to TEST1, it barely increased the damage. 

Note: The 500 RAP from hunter's mark should be considered both for the weapon damage portion (as RAP directly increases weapon damage, even though the RAP bonus from mark is not displayed on the character sheet) and the RAP coefficient portion, as if the hunter really had a 500 RAP buff. I cannot say that either portions are missing because the math doesn't check out anyhow, but something is not calculating as intended.

**Comment by Thrashattack** [comment]
Hunter Level 80 

No aspect no aura 

Bow level 1 and arrow level 1 ((5 ~ 10 ) +1.5 dmg)

RAP 429 (340 level + 89 talents)  

	ranged weapon damage +30.6 


Killshot expected damage = 896 ~906 {

	• Total weapon damage ( (5~10) + 1.5 + 30.6 ) =  37~42 
	• 200% weapon damage = 74 ~ 84 
	• Plus killshot damage = 822 
	• total = 896 ~ 906
}




### First test no mark no aspect 

    Target: Zanzil Zombie (http://www.wow-mania.com/armory/?npc=1488) 
	• Armor = 2.3 ~ 2.5 
	• Killshot Damage = 886 ~ 898 (Low)

#### If RAP 429 --> increases ranged weapon damage in 30.6 
#### then RAP 929(500+429) -->  should increase the RWD in ((30.6*929)/429)= 66.2

• Not ment to be showed in the sheet but to be calculated by the unit wich will recieve the damage. 



Killshot expected damage with mark = 966 ~977 {

	Total weapon damage ( (5~10) + 1.5 + 66.2 ) =  72,7~77,7 
	200% weapon damage = 144,4 ~ 155,4 
	Plus killshot damage = 822 
	total = 966 ~ 977
}



### Second Test With Mark

            Target: Zanzil Zombie (http://www.wow-mania.com/armory/?npc=1488) 
	        • Armor = 2.3 ~ 2.5 
	        • Killshot Damage = 956 ~966 (Low)




In conclusion i can see a problem with Kill Shot at all not only when the target is marked. 
Of course when the damage scales with items and stats and other talents (89 RAP from talents only 31 pts spent)
the difference betwen the values will increase aswell. 
The loss of dmg keep around 5% either to my tests or Dranah's.

**Comment by Dranah** [comment]
That's odd because as you can see my test values on non-marked targets were completely fine.
Do you have 100% arpen?

**Comment by Thrashattack** [comment]
The zoombie have 2.3 ~2.5 armor value and it was considered for the calculation. 

Considering i only tested in these zoombies with this variation of armor the damage reduction provided by the mobs armor to my lvl 80 hunter should be exatct 0.00034260655063 points of damage (case 2.5 armor). 
Therefore the difference of 10 points of damage between the expected and the obtained lead me to think that killshot isnt doing the correct damage. Or is just that multiplicative scale thing.

**Comment by Keader** [comment]
Test now please.

**Comment by Dranah** [comment]
TEST after fix: hunter with 4340 RAP against marked target (so effective RAP 4840)
Weapon damage is still 2022-2311 with 4830 effective RAP
Kill shot damage= 200% weapon damage + 650 (base) + 0,4 x RAP
**Expected damage: 6020-6544
Tested damage: 6106-6605**

The values are now very close but don't really match. 

So i've made another test with a low gear, 0 arpen hunter, 1223 RAP +mark vs 1723 RAP:

TEST1 (1223 RAP against unmarked) - Weapon damage 392-412
Expected damage: 1130-1154
Tested Damage: 1132-1154

TEST 2 (1723 RAP against unmarked) - Weapon damage 478-498
Expected damage: 1349- 1373
Tested damage: 1351-1373

TEST 3 (1223 RAP against marked - effective RAP 1723) - Weapon damage 478-498
Expected damage: 1349-1373 (same as test 2)
Tested damage: 1426-1448

So something is still calculating wrongly, adding a little extra damage to the intended.

**Comment by Dranah** [comment]
It seems this cannot be improved any further. This fix is closest to the intended damage so will stay this way.
At high gear levels it appears to be a negligible 1% difference anyway.
